### ISSUE TITLE
Java8 type-use annotations on qualified types not supported

### ISSUE THREAD

**Issue by RocusHalbasch** [checkstyle/checkstyle#2079]
Checkstyle fails to parse some code using type annotations on qualified types:

CheckstyleFail.java:
```java
import java.awt.geom.Rectangle2D;
import java.lang.annotation.ElementType;
import java.lang.annotation.Target;

public class CheckstyleFail {
	/* Causes parse failure */
	Rectangle2D.@Ann Double rect = null;
	
	/* Causes parse failure */
	public final Rectangle2D.@Ann Double getRect1() {
		return new Rectangle2D.Double();
	}

	/* Causes parse failure */
	public final Rectangle2D.Double getRect2() {
		return new Rectangle2D.@Ann Double();
	}

	/* Amazingly does not cause parse failure */
	public final Rectangle2D.Double getRect3() {
		Rectangle2D.@Ann Double rect = null;
		return rect;
	}
}

@Target({ ElementType.TYPE_USE })
@interface Ann {}
```
checkstyle.xml:
```XML
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
	<module name="TreeWalker" />
</module>
```

This is similar to bug [#904](https://github.com/checkstyle/checkstyle/issues/904): Java8 type-use annotations on arrays not supported

**Comment by romani** [comment]
fixed by #2891

**Comment by romani** [comment]
@RocusHalbasch  , 
as soon as https://travis-ci.org/checkstyle/checkstyle/jobs/108286542 is finished , binaries will be at 
https://oss.sonatype.org/content/repositories/snapshots/com/puppycrawl/tools/checkstyle/6.16-SNAPSHOT/ . Please test your sources to make sure that did not miss anything. Feel free to open new issues.
